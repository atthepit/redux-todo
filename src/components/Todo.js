import React from 'react'

const Todo = ({ todo, onClick }) => {
  const active = { textDecoration: todo.complete ? 'line-through' : 'none' }
  return (
    <div onClick={ () => onClick(todo.id) } style={ active }>
      { todo.text }
    </div>
  )
}

Todo.prototype.propTypes = {
  todo: React.PropTypes.shape({
    id: React.PropTypes.number.isRequired,
    title: React.PropTypes.string,
    complete: React.PropTypes.bool,
  }).isRequired,
  onClick: React.PropTypes.func,
}

export default Todo
