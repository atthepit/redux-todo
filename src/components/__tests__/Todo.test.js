import React from 'react'
import Todo from '../Todo'
import renderer from 'react-test-renderer'

describe('Test Todo component', () => {
  it('renders correctly', () =>{
    const props = {
      todo: { id: 0, text: 'Hello world', complete: false },
      onClick: jest.fn(),
    }

    const tree = renderer.create(
      <Todo { ...props } />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
