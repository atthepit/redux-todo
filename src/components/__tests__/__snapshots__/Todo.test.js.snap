exports[`Test Todo component renders correctly 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }>
  Hello world
</li>
`;
